pkg/sharding/listwatch.go

```diff
@@ -0,0 +1,88 @@
+package sharding
+
+import (
+	"crypto/md5"
```

Comment:
Why use a `crypto` hash function in this context? Wouldn't one of https://golang.org/pkg/hash/ be faster?